src/ts/metatxhandler.ts

```diff
@@ -49,379 +29,140 @@ export enum ChainID {
  * replay protection.
  */
 export class MetaTxHandler {
```

Comment:
This class is basically responsible for creating Forwarder objects. The logic inside this class should be moved a ForwarderFactory class that's responsible for creating forwarders. You can keep the MetaTxHandler class if you want, but all the logic inside it should be moved to the ForwarderFactory and just called by the MetaTxHandler.

There should be a ForwarderFactory abstract class, with the function `async createForwarder()` on it. Do the same process as you did for refactoring the forwarders themselves: Make a ProxyAccountForwarderFactory and RelayHubForwarderFactory move all the code into those factories then, start moving the shared code down into the base ForwarderFactories. Some shared code that I can already see will go on the base class would be:
```
getReplayProtection(replayProtectionAuth: ReplayProtectionType, signer: Wallet, proxyAddress: string) {
    switch {
       case ReplayProtectionType.BITFLIP: return new BitFlip(signer, proxyAddress);
       case Repla......
    }
}
```

Reply:
I created a protected getReplayProtection() function to simplify and renamed metaTxHandler to ForwarderFactory. It seems a bit overkill atm to make ProxyAccountForwarderFactory etc since there are only 4-5 functions and they are relatively simple overall.